packages/react-vapor/src/components/unsavedChangeModal/UnsavedChangeModalProvider.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import {Button} from '../button/Button';
+import {ModalComposite} from '../modal/ModalComposite';
+
+const defaultModalTitle = 'Unsaved Changes';
+const defaultModalClasses = ['mod-prompt', 'mod-fade-in-scale'];
+const defaultModalDescription =
+    'Are you sure you want to leave this page without saving? All unsaved changes will be lost.';
+const defaultConfirmButtonText = 'Exit without applying changes';
+
+export interface IUnsavedChangesModalProviderProps {
+    isDirty: boolean;
+    modalTitle?: string;
+    className?: string[];
+    children: (props: {promptBefore: (callbackOnDiscard: () => any) => boolean}) => React.ReactNode;
+    unsavedChangesDescription?: string;
+    confirmButtonText?: string;
+}
+
+export const UnsavedChangesModalProvider: React.FunctionComponent<IUnsavedChangesModalProviderProps> = ({
+    isDirty,
+    children,
+    modalTitle = defaultModalTitle,
+    className = defaultModalClasses,
+    unsavedChangesDescription = defaultModalDescription,
+    confirmButtonText = defaultConfirmButtonText,
+}) => {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [confirm, setConfirm] = React.useState(null);
+
+    const promptBefore = (callbackOnDiscard: () => any): boolean => {
```

Comment:
What's the use case for returning a boolean here?

Reply:
here: https://github.com/coveo/react-vapor/pull/1372/files#diff-b25018ab974a2628440b0e206a2d3c37R158

if the modal isn't dirty upon closing, the `promptBefore` will return true and execute the ` && handleClose()`